src/TaskLayout.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Various rendering options for a query.
+ */
+export class LayoutOptions {
+    hideTaskCount: boolean = false;
+    hideBacklinks: boolean = false;
+    hidePriority: boolean = false;
+    hideStartDate: boolean = false;
+    hideScheduledDate: boolean = false;
+    hideDoneDate: boolean = false;
+    hideDueDate: boolean = false;
+    hideRecurrenceRule: boolean = false;
+    hideEditButton: boolean = false;
+    hideUrgency: boolean = true;
+    shortMode: boolean = false;
+}
+
+export type LayoutComponent =
+    | 'description'
+    | 'priority'
+    | 'recurrenceRule'
+    | 'startDate'
+    | 'scheduledDate'
+    | 'dueDate'
+    | 'doneDate'
+    | 'blockLink';
+
+/**
+ * This represents the desired layout of tasks when they are rendered in a given configuration.
+ * The layout is used when flattening the task to a string and when rendering queries, and can be
+ * modified by applying LayoutOptions objects.
```

Comment:
```suggestion
  * modified by applying {@link LayoutOptions} objects.
```

Reply:
It helps navigating in WebStorm if symbol names in JSdoc are @linked... (though the @link is not needed in @see lines).

Reply:
I can try to add them manually, just note that I don't have these features to test in my own IDE (which is `nvim`), so it's a hit or miss :shrug:

Reply:
You can always just commit my suggested edit above, I think - perhaps after merging and pushing my branch.

But it's really easy for me to add them, so no worries at all if you don't link them.